### DISCUSSION TITLE
Firebase Core - Switch environments on the fly

### DISCUSSION THREAD

**Discussion by ollyde** [firebase/flutterfire#5626] [Feature request]
Just like the JS version it would be super handy if we could switch environments say (dev/prod) on the fly.
There is an implementation of FirebaseApp.config but it's disabled during runtime. Not sure why.

At the moment it's really difficult to determine on TestFlight/Google Play etc what is the "dev/prod" build using Flutter and building twice is annoying.

**Comment by kroikie** [comment]
Are you trying to point to a different project on the fly? The current [dev version of core](https://pub.dev/packages/firebase_core/versions/0.5.0-dev.2) does allow you to initialize different apps then you can determine which FirebaseApp is used when retrieving instances of other plugins like `firestore` for example:

```await Firebase.initializeApp(
    name: 'SecondaryApp',
    options: const FirebaseOptions(
        appId: 'my_appId',
        apiKey: 'my_apiKey',
        messagingSenderId: 'my_messagingSenderId',
        projectId: 'my_projectId'
    )
);
```

```FirebaseApp secondaryApp = Firebase.app('SecondaryApp');

Firestore firestore = Firestore.instanceFor(
  app: secondaryApp
);

firestore
  .collection('users')
  .snapshots()
  .listen((QuerySnapshot snapshot) {
     // Query snapshot of the users collection on the SecondaryApp
  });
```

**Reply by jamesdixon** [reply]
Question: does initializing the Firebase app using this method negate the need for the service files (plist/json) files?

**Comment by ollyde** [comment]
@kroikie thanks; is there any ETA when that would be in production? We cannot use Flutter dev for production pipelines :-)

This would be really really awesome if it worked in Prod, being able to switch environments is key for testers. Installing different builds and making multiple builds for 1 app (imagine 5 builds for 1 version..) is just miserable corporate experience.

**Comment by ollyde** [comment]
Having some issues with the dependencies

At the moment I've got:

```
  firebase_core: ^0.5.0-dev.2
  cloud_firestore: ^0.14.0-dev.1
  firebase_auth: ^0.17.0-dev.2
  firebase_messaging: ^7.0.0-dev.1
  firebase_dynamic_links: ^0.5.0+11
  firebase_analytics: ^6.0.0-dev.1
```

But there are tons of errors with the pods. (cleaned, updated, removed lock, etc)

Is there anything I'm supposed to do with these dev packages?

**Comment by darshankawar** [comment]
Hi @ollydixon,
There has been recent [updates](https://firebase.flutter.dev/) on Firebase plugins and I'd encourage you to update each of the listed above to latest and see if you are still getting the issues.
Thanks.

**Comment by ollyde** [comment]
@darshankawar thanks for the reply; I'm using the new versions now but there's some issues with the pods adding various build phases that break our build servers. Also it seems to be incompatible with Flutter 1.20.X

The switching doesn't seem to work either, probably the most important thing for us and Firebase :-)

**Comment by sima-11** [comment]
After a year, the switching still doesn't work. I logged a bug ticket 21 days ago and was closed by @darshankawar who suggested to find answers here.
During those 21 days I have posted a few follow up messages under my ticket but got no reply at all. 

This is frustrating, really.

**Reply by ollyde** [reply]
We decided to ditch Firebase cause of this and the extremely slow iOS build times. Using MongoDB now; never looking back (It's a gazillion times better). 🚀

**Comment by darshankawar** [comment]
Cc: @Salakar @Ehesp for further insights.

**Comment by ollyde** [comment]
We are now forced to use Firebase as mobile developers as Google has scrapped the nice and simple GCM for this 😢
